### PR TITLE
Run UI tests with @db_access / logins / pegasus on CircleCI

### PR DESCRIPTION
**Pull request by bcjordan**
- Fixes code.org resolving to studio.code.org on CircleCI
  - Sets pegasus domain to localhost.code.org:3000
- Fixes session persistence when running on CircleCI

### CODE REVIEW

**Comment by bcjordan** [comment]
While this appears to now be passing :tada:, the total run time of the tests has increased from 21:24 to 45:55—which might suggest we'd want to either experiment with (1) increasing parallelization or (2) separating a chunk of the tests into a third CircleCI container to avoid bloating our test run times.

(Or we may want to choose to merge first instead just to avoid overhead of having surprised on the test machine.) @islemaster I will leave that call up to you this coming week as I won't be around to manage any impact/possible revert.

+@wjordan for reviewer as he has some more context on a few of the fixes required to get sessions & pegasus/dashboard subdomains working.

File: Gemfile [hunk 1]
```diff
@@ -135,7 +135,7 @@ gem 'newrelic_rpm', '~> 3.10.0.279', group: [:staging, :production] # perf/error
 
 gem 'redcarpet', '~> 3.3.4'
 
-gem 'geocoder'
+gem 'geocoder', github: 'wjordan/geocoder', ref: 'rack-request-fix'
```
Comment (bcjordan) [inline Gemfile:138]: We may want to merge in https://github.com/code-dot-org/code-dot-org/pull/10140 separately from this just for separate merge monitoring/rollback if anything comes up. This only affected ./bin/dashboard-server & caused JS and server log errors on the frontpage when running in that mode. Possibly only cosmetic as far as CircleCI UI tests are concerned.

**Comment by bcjordan** [comment]
Re: integrating this branch, since the "test" branch has any changes manually accepted for the test machine's branch and staging may not, we may also need to merge eyes baselines test -> staging before merging this in for things to go most smoothly.

**Comment by bcjordan** [comment]
Another thought on potentially speeding this up—we could (1) split up the applab/eyes.feature and eyes.feature files into much smaller files to likely cut down ~15 minutes (they take 12 and 21 minutes each, and the rest are <5), and (2) possibly run eyes tests in parallel with normal tests

```
The key is circle/19735/ChromeLatestWin7_applab_eyes_eyes_output.html
UI tests for ChromeLatestWin7_applab_eyes_eyes succeeded (12:13 minutes, 9 passed)
The key is circle/19735/ChromeLatestWin7_eyes_eyes_output.html
UI tests for ChromeLatestWin7_eyes_eyes succeeded (21:37 minutes, 36 passed)
```

**Comment by islemaster** [comment]
I'm not in favor of merging this while it more than doubles run-time.  Splitting up the eyes feature files seems like a good low-hanging fruit though; I'll look for time to try this out.

**Comment by bcjordan** [comment]
Merged in https://github.com/code-dot-org/code-dot-org/pull/10311 to test timing changes

**Pull request merged**